api/scanner/media_encoding/executable_worker/executable_worker_test.go

```diff
@@ -41,3 +42,35 @@ func setEnv(key, value string) func() {
 		os.Setenv(key, org)
 	}
 }
+
+func TestInitFfprobePath(t *testing.T) {
+	t.Run("PathFail", func(t *testing.T) {
+		err := executable_worker.InitFfprobePath()
+		if err == nil {
+			t.Fatalf("InitFfprobePath() returns nil, want an error")
+		}
+	})
+
+	t.Run("VersionFail", func(t *testing.T) {
+		donePath := setPathWithCurrent("./testdata/bin")
+		defer donePath()
+
+		doneEnv := setEnv("FAIL_WITH", "expect failure")
```

Comment:
I don't understand here. `ffprobe` checks this `FAIL_WITH` ENV and returns non-zero if it's not empty. It's not a go testing framework thing.

`PathFail` in line 47 already checks no `ffprobe` case, since it doesn't set `PATH`.

It's a unit-test, but not a integrated test. It's better to mock the dependency and test the code, but not dependency functions.

Reply:
Ah, ok. I misunderstood the executable: I thought it is real, while in fact it was a stub SH script, which is part of this PR as well...

> It's a unit-test, but not a integrated test. It's better to mock the dependency and test the code, but not dependency functions.

Sure. That is true. However, in general, do you see a value in having key integration tests per functionality (in this case, per worker)? Under such tests I mean a code, which tries to process some prepared examples of media files and verifies intermediate and final results.

Reply:
Yes, of course. I have two direction:

1. Test dependencies with dependencies images, to verify features we want being covered by them.
2. Integration tests on the final image (may with some other tools) for end 2 end tests.

The first one seems easier, but the second one is completeness. I don't know how hard the second is now.

Reply:
As the `dependencies` image is not usable and not shipable, I don't see practical value in the 1st way. Also, it doesn't look like integration tests.
I like the 2nd way more, as here we could test the product better, as everything is integrated and ready.

Reply:
Filed https://github.com/photoview/photoview/issues/1057